debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java

```diff
@@ -91,6 +105,55 @@
     public static final Field SERVER_NAME = RelationalDatabaseConnectorConfig.SERVER_NAME
             .withValidation(CommonConnectorConfig::validateServerNameIsDifferentFromHistoryTopicName);
 
+    public static final Field CONNECTOR_ADAPTER = Field.create(DATABASE_CONFIG_PREFIX + "connection.adapter")
+            .withDisplayName("Connector adapter")
+            .withEnum(ConnectorAdapter.class, ConnectorAdapter.XSTREAM)
+            .withWidth(Width.MEDIUM)
+            .withImportance(Importance.HIGH)
+            .withDescription("There are two adapters: XStream and LogMiner.");
+
+    public static final Field SNAPSHOT_SKIP_LOCKS = Field.create("snapshot.skip.locks")
+            .withDisplayName("Should schema be locked as we build the schema snapshot?")
+            .withType(Type.BOOLEAN)
+            .withWidth(Width.SHORT)
+            .withImportance(Importance.LOW)
+            .withDefault(false)
+            .withValidation(Field::isBoolean)
+            .withDescription(
+                    "If true, locks all tables to be captured as we build the schema snapshot. This will prevent from any concurrent schema changes being applied to them.");
+
+    public static final Field LOG_MINING_STRATEGY = Field.create("log.mining.strategy")
+            .withDisplayName("Log Mining Strategy")
+            .withEnum(LogMiningStrategy.class, LogMiningStrategy.CATALOG_IN_REDO)
+            .withWidth(Width.MEDIUM)
+            .withImportance(Importance.HIGH)
+            .withDescription("There are strategies: Online catalog with faster mining but no captured DDL. Another - with data dictionary loaded into REDO LOG files");
+
+    // this option could be true up to Oracle 18c version. Starting from Oracle 19c this option cannot be true todo should we do it?
+    public static final Field CONTINUOUS_MINE = Field.create("log.mining.continuous.mine")
+            .withDisplayName("Should log mining session configured with CONTINUOUS_MINE setting?")
+            .withType(Type.BOOLEAN)
+            .withWidth(Width.SHORT)
+            .withImportance(Importance.LOW)
+            .withDefault(false)
+            .withValidation(Field::isBoolean)
+            .withDescription("If true, CONTINUOUS_MINE option will be added to the log mining session. This will manage log files switches seamlessly.");
+
+    public static final Field SNAPSHOT_ENHANCEMENT_TOKEN = Field.create("snapshot.enhance.predicate.scn")
+            .withDisplayName("A string to replace on snapshot predicate enhancement")
+            .withType(Type.STRING)
+            .withWidth(Width.MEDIUM)
+            .withImportance(Importance.HIGH)
+            .withDescription("A token to replace on snapshot predicate template");
+
+    public static final Field DRIVER_TYPE = Field.create("database.driver.type")
```

Comment:
If there is hard dependency between `CONNECTOR_ADAPTER` and `DRIVER_TYPE` then it should either be controlled by a single config param or at least there should be a validator rejecting invalid combinations.

Reply:
I went ahead and removed this configuration property and rolled this behavior into the enum class with `#getDriverType()`.